### PR TITLE
IIGO Negative Allocation. Quick fix

### PR DESCRIPTION
**Pull request by E-Mac27**
# Summary
As title.

# Tests
All test pass

### CODE REVIEW

**Comment by Neelesh99** [comment]
Looks good, should we also put checks on tax and sanction payments. Just incase?

**Review by lhl2617 — APPROVED** [review]
image


certified negative

**Comment by E-Mac27** [comment]
> Looks good, should we also put checks on tax and sanction payments. Just incase?

I've ran the sim for 1500 turns and this didn't cause an issue but I dont mind being more cautious and adding it.

Edit: Only allocation throws an error for some reason. Sanction and Tax just logs and assigns 0 value. So checks are already in place.

File: internal/server/iigo.go [hunk 1]
```diff
@@ -90,7 +90,10 @@ func (s *SOMASServer) runIIGOAllocations() error {
 	allocationMap := make(map[shared.ClientID]shared.Resources)
 	for clientID, v := range clientMap {
 		allocation := v.RequestAllocation()
-
+		if allocation < 0 {
+			s.logf("Negative allocation of %v by %v. Changing allocation to 0", allocation, clientID)
```
Comment (lhl2617) [inline internal/server/iigo.go]: let's save some time

```suggestion
		if allocation < 0 || math.IsNaN(allocation) {
			s.logf("Invalid allocation of %v by %v. Changing allocation to 0", allocation, clientID)
```

**Pull request merged**